src/dq_suite/output_transformations.py

```diff
@@ -347,47 +389,102 @@ def get_single_validation_result_dict(
         get_parameters_from_results(result=expectation_result)
     )
 
+    rule_name = expectation_result["expectation_config"]["meta"]["rule_name"]
     return {
         "aantalValideRecords": valid_records,
         "aantalReferentieRecords": total_count,
         "percentageValideRecords": percentage_of_valid_records,
         # TODO/check: rename dqDatum, discuss all field names
         "dqDatum": run_time,
         "dqResultaat": validation_result,
-        "regelNaam": humps.pascalize(expectation_type),
+        "regelNaam": rule_name,
+        "regelParameters": validation_parameters,
+        "bronTabelId": table_id,
+    }
+ 
+ 
+def get_custom_validation_results(
+    expectation_result: dict, run_time: datetime, table_id: str, df: DataFrame
+) -> dict:
+    observed_str = expectation_result["result"].get("observed_value", "")
+
+    # Catch the number with regex
+    match = re.match(r"(\d+)", observed_str)
+    unexpected_count = int(match.group(1)) if match else 0
+
+    total_count = df.count()
+
+    percentage_of_valid_records = (
+        (total_count - unexpected_count) / total_count
+        if total_count and total_count > 0
```

Comment:
In what case could `total_count` be empty? (Since it is defined as `df.count()` above.)